devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -363,8 +366,10 @@ public void execute() throws MojoFailureException, MojoExecutionException {
 
         initToolchain();
 
-        //we always want to compile if needed, so if it is run from the parent it will compile dependent projects
+        // we always want to compile if needed, so if it is run from the parent it will compile dependent projects
         handleAutoCompile();
+        // If the jandex plugin is configured, we want to index classes.
+        initClassIndexes();
```

Comment:
I think one of the workarounds would be to move the `jandex` goal to the `compile` phase. For projects doing that, the `jandex` plugin now will be executed twice. I guess one of the quick fixes would be to check whether the `META-INF/jandex.idx` already exists in the `project.getBuild().getOutputDirectory()` and if it does not and the jandex plugin is configured we can execute it.
If the existing `META-INF/jandex.idx` isn't up-to-date, currently, it's not a big deal. We re-indexing reloadable dependencies during bootstrap anyway, at the moment.